src/Samples/Sarif.WorkItems.Sample/ProgramStubs.cs

```diff
@@ -0,0 +1,233 @@
+        }
+
+        private class SarifLog
+        {
+            public IList<Run> Runs { get; internal set; }
+
+            internal static IEnumerable<WorkItemFilingMetadata> CreateWorkItemFilingMetadataFromLogs(IEnumerable<SarifLog> sarifLogsToFile)
+            {
+                throw new NotImplementedException();
+            }
+
+            public class Run
+            {
+                public IList<Result> Results { get; internal set; }
+            }
+        }
+
+        private class SarifLogBaselinerFactory
+        {
+            internal static ISarifLogBaseliner CreateSarifLogBaseliner(object standard)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private class SarifBaselineType
+        {
+            public static object Standard { get; internal set; }
+        }
+
+        private class SarifSplitter
+        {
+            internal static IEnumerable<SarifLog> Split(SarifLog sarifLog, Func<Result, bool> filterPredicate, SplittingStrategy splittingStrategy)
+            {
+                throw new NotImplementedException();
+            }
+
+            internal class Result
+            {
+                public BaselineState BaselineState;
+            }
+
+            internal class SarifSplittingStrategy
+            {
+            }
+        }
+
+        private enum BaselineState
+        {
+            New
+        }
+
+        private enum SplittingStrategy
+        {
+            OneFilePerRunPerRule
+        }
+
+        private class WorkItemFilingMetadata
+        {
+            public string Title { get; internal set; }
+
+            public string AreaPath { get; internal set; }
+
+            public string Body { get; internal set; }
+
+            public HashSet<string> Tags { get; internal set; }
+
+            public Uri Uri { get; internal set; }
+
+            public List<object> Attachments { get; internal set; }
+        }
+
+
+        private static void LogFailure(Exception exception)
+        {
+            throw new NotImplementedException();
+        }
+
+        private class AdoWorkItemFiler
+        {
+            public AdoWorkItemFiler()
+            {
+            }
+
+            public bool FileAttachments { get; internal set; }
+
+            public event EventHandler<WorkItemFiledOrUpdatedArgs> WorkItemFiledOrUpdated;
+
+            internal async Task FileWorkItems(IEnumerable<WorkItemFilingMetadata> workItemsToFile)
+            {
+                throw new NotImplementedException();
+            }
+
+            public class WorkItemFiledOrUpdatedArgs
+            {
+                public WorkItemFilingMetadata Metadata;
+            }
+        }
+
+        private class Result
+        {
+            public string Guid { get; internal set; }
+            public IList<Uri> WorkItemUris { get; internal set; }
+        }
+
+        private class GuidMappingVisitor
```

Comment:
>GuidMappingVisitor [](start = 22, length = 18)

Derived from `SarifRewritingVisitor`, I assume -- worth stating in the code here.

Reply:
This is throwaway code, as mentioned in the description and at the top of this file, this content was not intended to be reviewed. Instead, my hope was that people would focus on the shape of the API as it is used. This is the focus of this exercise.

---
In reply to: [307083504](https://github.com/microsoft/sarif-sdk/pull/1589#discussion_r307083504) [](ancestors = 307083504)